daft/io/_huggingface.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from daft.api_annotations import PublicAPI
+
+from ._parquet import read_parquet
+
+if TYPE_CHECKING:
+    from daft.daft import IOConfig
+    from daft.dataframe import DataFrame
+
+
+@PublicAPI
+def read_huggingface(repo: str, io_config: IOConfig | None = None) -> DataFrame:
+    """Create a DataFrame from a Hugging Face dataset.
+
+    Currently only supports Parquet datasets.
```

Comment:
This is incorrect. It should support any dataset that is eligible for parquet auto conversion https://huggingface.co/docs/dataset-viewer/en/parquet, which according to huggingface, _should_ be all public datasets. 

> Datasets can be published in any format (CSV, JSONL, directories of images, etc.) to the Hub, and they are easily accessed with the 🤗 [Datasets](https://huggingface.co/docs/datasets/) library. For a more performant experience (especially when it comes to large datasets), the dataset viewer automatically converts every dataset to the [Parquet](https://parquet.apache.org/) format.

Reply:
Ah right. Let me change that